crates/telio-firewall/src/firewall.rs

```diff
@@ -468,7 +468,6 @@ impl StatefullFirewall {
                 *tx_alive = false;
                 if !*rx_alive {
                     telio_log_trace!("Removing TCP conntrack entry {:?}", e.key());
-                    e.remove();
```

Comment:
Correct me if I'm wrong, but If the connection is not removed, and the connection is marked as not remote_initiated(meaning our side created the connection), the remote will send new SYN and new tcp connection will be made, but our firewall will think we created that connection. This could maybe be fixed by checking rx_alive to filter received packets, but not updating cache timeout, so the cache entry will just expire eventually.

Reply:
Maybe when `!tx_alive && !rx_alive && !conn_remote_initiated` we should not update ttl and remove the entry when we receive `ack`?